### PR TITLE
Add max argument overload to yargs.demand

### PR DESCRIPTION
**Pull request by elisee**
Please fill in this template.
- [x] Prefer to make your PR against the `types-2.0` branch.
- [x] The package does not provide its own types, and you can not add them.
- [x] Test the change in your own code.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).

If changing an existing definition:
- [x] Provide a URL to  documentation or source code which provides context for the suggested changes: https://github.com/yargs/yargs#demandcount-max-msg
- [x] Increase the version number in the header if appropriate. (Not needed.)

### CODE REVIEW

**Comment by poelstra** [comment]
@elisee According to the docs, the `msg` argument can be optional. Can you also please add some test cases?

**Comment by elisee** [comment]
@poelstra There we go.

File: yargs/index.d.ts [hunk 1]
```diff
@@ -40,6 +40,8 @@ declare namespace yargs {
         demand(keys: string[], required?: boolean): Argv;
         demand(positionals: number, required?: boolean): Argv;
         demand(positionals: number, msg: string): Argv;
+        demand(positionals: number, max: number): Argv;
+        demand(positionals: number, max: number, msg: string): Argv;
```
Comment (mizunashi-mana) [inline yargs/index.d.ts]: Can rewrite ;)

```typescript
demand(positionals: number, max: number, msg?: string): Argv;
```

**Comment by andy-ms** [comment]
Yargs tests are currently failing (fixed by #12344); please merge changes from that once it's in.

**Comment by elisee** [comment]
@mizunashi-mana done!

**Comment by elisee** [comment]
Also merged the fix as @andy-ms requested.

**Comment by poelstra** [comment]
@elisee Nice, thnx!

:+1: from me

**Pull request merged**